rpcd/playbooks/test-maas.yml

```diff
@@ -0,0 +1,75 @@
+---
+# Copyright 2015, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- hosts: hosts
+  tasks:
+    - name: Create fake hpasmcli
+      copy:
+        src: /opt/rpc-openstack/maas/testing/fake_hp_monitoring.py
+        dest: /usr/local/bin/hpasmcli
+        mode: 0755
+      tags:
+        - setup
+
+    - name: Create fake hpssacli
+      copy:
+        src: /opt/rpc-openstack/maas/testing/fake_hp_monitoring.py
+        dest: /usr/local/bin/hpssacli
+        mode: 0755
+      tags:
+        - setup
+
+    # This attempts to work around the frequent memory errors with the agent
+    - name: Check if raxmon agent connected
+      shell: sleep 300; raxmon-agent-connections-list --agent-id {{ inventory_hostname }}{{ maas_fqdn_extension | default('') }} | grep -qv '\[]'; if [ $? -eq 0 ]; then if [ "$(date '+%s')" -gt "$(sleep 5; date -r /var/log/rackspace-monitoring-agent.log '+%s')" ]; then restart rackspace-monitoring-agent; else echo okay; fi; else restart rackspace-monitoring-agent; fi;
```

Comment:
So there's a bit of a fudge factor with this number. When I reviewed some of MaaS logs it looked like it took ~ 75 seconds for all the plugin checks to be configured and run once. This needs to happen so that all the metric data is uploaded to MaaS, specifically the names of the metrics. So, lets say 80 seconds is probably the minimum the pause can be. To detect if the agent is no longer connected I both interrogate the API and check when the log file was last updated. I'm not sure what the timeout on the API is, but when I was testing just checking this after a couple of minutes was not sufficient to detect failures. If the logs were updated for 80 seconds after the agent was started that should be good enough unless the last thing in the logs is an error. I've essentially avoided parsing the logs by putting in a longer time out, from what I saw of the logs any error should have already happened by that time. For the most part the additional overhead should only ever work out to be a few minutes, probably less that 10 minutes, on top of the minimum it would need to wait.